.github/workflows/upload-test-stats.yml

```diff
@@ -98,7 +98,10 @@ jobs:
           WORKFLOW_RUN_ATTEMPT: ${{ github.event.workflow_run.run_attempt }}
           WORKFLOW_URL: ${{ github.event.workflow_run.html_url }}
           HEAD_REPOSITORY: ${{ github.event.workflow_run.head_repository.full_name }}
-          HEAD_BRANCH: ${{ github.event.workflow_run.head_branch }}
+          # head_branch is empty for tag-based runs (e.g., trunk/{sha})
+          # Fallback to a synthesized trunk/{head_sha} tag so downstream logic
+          # can detect trunk restarts and upload full test_run data.
+          HEAD_BRANCH: ${{ github.event.workflow_run.head_branch || format('trunk/{0}', github.event.workflow_run.head_sha) }}
```

Comment:
I don't like this, could be brittle. Any suggestion on how to actually detect that this was run on a tag?